plat_windows.go

```diff
@@ -396,6 +397,20 @@ func (task *TaskRun) prepareCommand(index int) *CommandExecutionError {
 	return nil
 }
 
+// Set an environment variable in each command.  This can be called from a feature's
+// NewTaskFeature method to set variables for the task.
+func (task *TaskRun) setVariable(variable string, value string) error {
```

Comment:
This is just a nit, not a required change.

Nit: I would have probably called this `setEnvironmentVariable` for clarity, and made it a function of _the environment_ rather than the task, but it doesn't really matter. This avoids the duplication of the loop over the task commands in both platform implementations. If you provide a method that just updates an environment block in a platform-independent way, you can have a single loop over the task commands, that is platform independent. It doesn't really matter though.

Note, originally the `windows` and `all-unix-style` process implementations were quite different, but with go 1.10 it became possible to use the standard library for running processes as a different user under windows, at which point it became possible to have very similar implementations for both. I intend to clean up some of the duplication in a future PR, and at that point will probably introduce a separate go type for an environment block (something like `type Environment []string`) that will have its own `func (env *Environment) Set(envVar string, value string)`, `func (env *Environment) Clear(envVar string)` functions for modifying it. So it makes sense to rework this at the same time, so we can leave it as is for now.

Reply:
I'll leave that to you :)